Doc/library/asyncio-policy.rst

```diff
@@ -0,0 +1,219 @@
+.. currentmodule:: asyncio
+
+
+.. _asyncio-policies:
+
+========
+Policies
+========
+
+Event loop management is controlled by an event loop policy, which
+is a global per-process object. There is a default policy and an
+API to change the policy. A policy defines the notion of context;
+a policy manages a separate event loop per context. The default
+policy's notion of context is defined as the current thread.
+
+asyncio allows to customize how :func:`get_event_loop`,
+:func:`set_event_loop`, and :func:`new_event_loop` functions behave
+by using a custom event loop policy.
+
+A policy is a special object implementing APIs defined in the
+:class:`AbstractEventLoopPolicy` abstract base class.
+
+
+Accessing Policies
+==================
+
+The following functions can be used to get and set the current
+policy for the current process:
+
+.. function:: get_event_loop_policy()
+
+   Return the current process-wide policy.
+
+.. function:: set_event_loop_policy(policy)
+
+   Set the current process-wide policy to *policy*.
+
+   If *policy* is ``None``, the default policy is restored.
+
+
+Policy Objects
+==============
+
+The abstract event loop policy base class is defined as follows:
+
+.. class:: AbstractEventLoopPolicy
+
+   An abstract base class for asyncio policies.
+
+   .. method:: get_event_loop()
+
+      Get the event loop for the current context.
+
+      Return an event loop object implementing the
+      :class:`AbstractEventLoop` interface.
+
+      This method should never return ``None``.
+
+      .. versionchanged:: 3.6
+
+   .. method:: set_event_loop(loop)
+
+      Set the event loop for the current context to *loop*.
+
+   .. method:: new_event_loop()
+
+      Create and return a new event loop object.
+
+      This method should never return ``None``.
+
+   .. method:: get_child_watcher()
+
+      Get a child process watcher object.
+
+      Return a watcher object implementing the
+      :class:`AbstractChildWatcher` interface.
+
+      This function is Unix specific.
+
+   .. method:: set_child_watcher(watcher)
+
+      Get the current child process watcher to *watcher*.
+
+      This function is Unix specific.
+
+
+asyncio ships with the following built-in policies:
+
+
+.. class:: DefaultEventLoopPolicy
+
+   The default asyncio policy.  Uses :class:`SelectorEventLoop`
+   on both Unix and Windows platforms.
+
+   There is no need to install the default policy manually; asyncio
+   is configured to use it automatically.
+
+
+.. class:: WindowsProactorEventLoopPolicy
+
+   An alternative event loop policy that uses the
+   :class:`ProactorEventLoop` event loop implementation.
+
+   Availability: Windows.
+
+
+Process Watchers
+================
+
+Process watchers allow to customize how event loops monitor
```

Comment:
A process watcher allows customization of how an event loop monitors child processes on Unix. Specifically, the event loop needs to know when a child process has finished its execution.